profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java

```diff
@@ -251,7 +252,26 @@ public Object getConnectionStatus(EntityReference ref, Map<String, Object> param
 		
 		return connectionsLogic.getConnectionStatus(uuid, parameters.get("friendId").toString());
 	}
-	
+
+    @EntityCustomAction(action="unreadMessagesCount",viewKey=EntityView.VIEW_SHOW)
+	public Object getUnreadMessagesCount(EntityReference ref) {
+
+		if(!sakaiProxy.isLoggedIn()) {
+			throw new SecurityException("You must be logged in to get the unread messages count.");
+		}
+
+		//convert input to uuid
+		String uuid = sakaiProxy.ensureUuid(ref.getId());
+		if(StringUtils.isBlank(uuid)) {
+			throw new EntityNotFoundException("Invalid user.", ref.getId());
+		}
+        
+		if (!sakaiProxy.getCurrentUserId().equals(uuid)) {
```

Comment:
I'd think if you're an admin user calling this, you could check any message count